specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,530 @@
+        }
+      },
+      "put": {
+        "tags": [
+          "WebApplicationFirewallPolicies"
+        ],
+        "description": "Creates or update policy with specified rule set name within a resource group.",
+        "operationId": "WebApplicationFirewallPolicies_CreateOrUpdate",
+        "x-ms-examples": {
+          "Creates or updates a WAF policy within a resource group": {
+            "$ref": "./examples/WafPolicyCreateOrUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "policyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the policy."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "description": "Policy to be created.",
+            "in": "body",
+            "name": "parameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/WebApplicationFirewallPolicy"
+            }
+          },
+          {
+            "$ref": "network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/WebApplicationFirewallPolicy"
+            }
+          },
+          "201": {
+            "description": "Created. The request has been fulfilled and a new protection policy has been created.",
+            "schema": {
+              "$ref": "#/definitions/WebApplicationFirewallPolicy"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "WebApplicationFirewallPolicies"
+        ],
+        "description": "Deletes Policy",
+        "operationId": "WebApplicationFirewallPolicies_Delete",
+        "x-ms-examples": {
+          "Deletes a WAF policy within a resource group": {
+            "$ref": "./examples/WafPolicyDelete.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "policyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the policy."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```

Comment:
Please add a default error response

Reply:
I did the powershell testing, if add default error response, it will generate different exception which won't be caught. Please reference the applicationGateway.json delete operation in the network.